src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        validator.validateOperation(fruitTransaction);
+        String fruit = fruitTransaction.getFruit();
+        validator.validateFruit(fruit);
+        int amountToPurchase = fruitTransaction.getQuantity();
+        validator.validateAmount(amountToPurchase, FruitTransaction.Operation.PURCHASE);
+
+        Optional<Integer> currentBalance = Optional.ofNullable(Storage.fruitStorage.get(fruit));
```

Comment:
int currentBalance = Optional.ofNullable(Storage.fruitStorage.get(fruit)).orElse(0);
imo makes more sense to deal with Optional on one line, fix it in every handler